### PR TITLE
fix: MR-2542 Ensure that trailing spaces in email address inputs do not trigger errors

### PR DESCRIPTION
**Pull request by haworku**
## Summary
Looking for feedback on the approach. I tried addressing the bug in two ways
- Using `.trim` in yup validations. This ensures validation error is simply not raised. See 482e326ea69c9bc0f161dbfe9ab047410aca1fe5 
- Direct remove leading and trailing whitespace from the input value itself `onBlur` to prevent this class of issue from arising, at least where our app-wide components are used . See 6ccda52dc6f5825503d1c6d447da43b887eaad4d 

Other change in this PR is slimming contacts and actuary contacts JSX by using `<FieldTextInput />` to remove extraneous markdown. 

#### Related issues
https://qmacbis.atlassian.net/browse/MR-2542

## QA guidance
- Besides testing bugfix to email address inputs, worth running through contacts and actuary contacts UI, make sure I didn't cause a regression here. Seems like we are testing with unit tests and Cypress pretty thoroughly here which is good to see.

### CODE REVIEW

File: services/app-web/src/components/Form/FieldYesNo/FieldYesNo.tsx [hunk 1]
```diff
@@ -23,7 +23,7 @@ export type FieldYesNoProps = {
     hint?: React.ReactNode
     showError?: boolean
     id: string
-    variant?: 'PRIMARY' | 'SECONDARY' // secondary variant used for nested fields yes/no fields under an overarching heading
+    variant?: 'TOPLEVEL' | 'SUBHEAD' // subhead variant used for nested fields yes/no fields under an overarching heading
```
Comment (haworku) [inline services/app-web/src/components/Form/FieldYesNo/FieldYesNo.tsx:26]: @macrael FYI saw your comment after I merged but... your wish is granted 🪄 https://github.com/CMSgov/managed-care-review/pull/1390#discussion_r1053541129

**Review by rswerve — APPROVED** [review]
I like the onBlur functionality.  Will keep the spaces from causing any problems downstream.

I entered a bunch of contacts with leading and trailing spaces.  No problem on submit.

File: services/app-web/src/components/Form/FieldTextInput/FieldTextInput.tsx [hunk 2]
```diff
@@ -16,35 +18,59 @@ import { PoliteErrorMessage } from '../../'
...
+            [styles.nestedField]: variant === 'SUBHEAD',
+        },
+        className
+    )
+
+    // Latch into onBlur to do any input cleaning
+    // Initial use case is to trim away trailing whitespace in email addresses
+    const customOnBlur = (e: React.FocusEvent<HTMLInputElement, Element>) => {
+        if (!e) return
+        helpers.setValue(field.value.trim())
```
Comment (JasonLin0991) [inline services/app-web/src/components/Form/FieldTextInput/FieldTextInput.tsx:56]: This `customOnBlur` does not seem to be working for email input on the contacts page. When not focused on the email input, it still retains leading and trailing spaces. 

You can continue to supporting documents with leading trailing spaces, but **it does trim**, even when `trim()` is removed from the the Yup validation in `Contacts.tsx`. I think is a graphical issue only.

**Review by JasonLin0991 — CHANGES_REQUESTED** [review]
Looks good. just one issue with the email input.

**Review by JasonLin0991 — APPROVED** [review]
Awesome! Thanks for those changes.

**Pull request merged**